spec-main/api-dialog-spec.ts

```diff
@@ -121,6 +121,62 @@ describe('dialog module', () => {
     });
   });
 
+  describe('showMessageBox with signal', () => {
+    afterEach(closeAllWindows);
+
+    it('closes message box immediately', async () => {
+      const controller = new AbortController();
+      const signal = controller.signal;
+      const w = new BrowserWindow();
+      const p = dialog.showMessageBox(w, { signal, message: 'i am message' });
+      controller.abort();
+      const result = await p;
+      expect(result.response).to.equal(0);
+    });
+
+    it('closes message box after a while', async () => {
+      const controller = new AbortController();
+      const signal = controller.signal;
+      const w = new BrowserWindow();
+      const p = dialog.showMessageBox(w, { signal, message: 'i am message' });
+      await delay(500);
```

Comment:
Why 500ms here? Would 0ms work equally well? If not, is there some event we can listen for?

`delay(500)` smells very much like flake to me.

Reply:
On Windows it takes some time before we can know the dialog's HWND (the dialog runs in a new thread and the dialog HWND is created asynchronously), the delay here is to make the test run the code path that closes the window after HWND is available. 

Sometimes even 500ms is not enough for the HWND to be available, but this is fine as no matter which code path it runs the showMessageBox should be cancelled successfully.

Reply:
Hm, is there any other way we could detect this event? e.g. a blur event?

Reply:
Showing a a child dialog does not remove the focus of parent window, so browser window's `blur` event will not emit. The only way I can think of is to load a web page and setup a `window.onblur` listener in it to send the event, which needs to careful to be not flaky.

Waiting for fixed time is usually bad because it makes test flaky, but this test is not flaky no matter what the timeout is, it only needs to confirm dialog can be closed after fully opened, and I think `delay(500)` does the work well.

Reply:
Yeah, it won't necessarily make the test fail, but it does mean the test will test _different things_, undetectably, depending on how fast the underlying OS is. I'd prefer that our tests test the same thing every time if possible.

I won't block the PR on this concern though.